lambda/docker-runner.yaml

```diff
@@ -0,0 +1,202 @@
+          - PerformanceMode
+          - Encrypted
+          - EfsPath
+Parameters:
+  PerformanceMode:
+    Type: String
+    Default: "generalPurpose"
+    Description: The EFS disk performance mode
+    AllowedValues:
+      - generalPurpose
+      - maxIO
+  Encrypted:
+    Type: String
+    Default: true
+    Description: True to enable EFS encryption, false to disable.
+    AllowedValues:
+      - true
+      - false
+  EfsPath:
+    Type: String
+    Description: Path to access EFS file store.
+    Default: /mnt/efs
+  MemorySize:
+    Type: Number
+    MinValue: 1024
+    MaxValue: 10240
+    Default: 1024
+    Description: >
+      The amount of memory (in MB) available to the function at runtime.
+      Increasing the function's memory also increases its CPU allocation.
+  ImageUri:
+    Type: String
+    Description: >
+      The URI of the Amazon ECR repository for the Lambda function's container image.
+      Example: 111111111.dkr.ecr.us-east-1.amazonaws.com/MY-DOCKER:latest
+  EntryPoints:
+    Type: CommaDelimitedList
+    Default: ""
+    Description: (Optional) Comma separated list of entry points to docker container
+  Command:
+    Type: CommaDelimitedList
+    Default: ""
+    Description: (Optional) Comma separated list of parameters to pass in with docker ENTRYPOINT
+  WorkingDirectory:
+    Type: String
+    Default: ""
+    Description: (Optional) The working directory for the docker container
+  EnableSchedule:
+    Description: >
+      true to run on a schedule, false to disable. If enabled a valid Schedule must be provided
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+  Schedule:
+    Description: >
+      Schedule to execute the docker, can be a rate or a cron schedule. Format at
+      https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+    Type: String
+    Default: cron(0 09 ? * MON *)  # Run every Monday at 9am
+    ConstraintDescription: "Use schedule format: https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html"
+  SynapsePAT:
+    Type: String
+    NoEcho: true
+    Default: ""
+    Description: >
+      (Optional) The Synapse personal access token (PAT).
+      https://docs.synapse.org/rest/POST/personalAccessToken.html
+Conditions:
+  HasEntryPoints: !Not
+    - !Equals
+      - !Join [ "", !Ref EntryPoints ]
+      - ''
+  HasCommand: !Not
+    - !Equals
+      - !Join ["", !Ref Command]
+      - ''
+  HasWorkingDirectory: !Not [!Equals [!Ref WorkingDirectory, ""]]
+  HasSynapsePAT: !Not [!Equals [!Ref SynapsePAT, ""]]
+Resources:
+  FileSystemResource:
+    Type: 'AWS::EFS::FileSystem'
+    Properties:
+      PerformanceMode: !Ref PerformanceMode
+      Encrypted: !Ref Encrypted
+      FileSystemPolicy:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Action:
+              - "elasticfilesystem:ClientMount"
+            Principal:
+              AWS: "*"
```

Comment:
This allows any account principal to mount the EFS AccessPoint created for this lambda? I think there needs to be a condition here to limit EFS access to the intended lambda, or use filesystem permissions (ie. uid/gid) on EFS to manage access to the files stored on it.

Reply:
removed.  we don't need this since access to mount  filesystem is already provided to the lambda function in EfsPolicy